### PR TITLE
LPD-40974 Inconsistent Deprecation Feature Flag Behavior in New Installations

### PR DESCRIPTION
**Pull request by thiago-buarqque**
https://liferay.atlassian.net/browse/LPD-40974

# Issue
Our `FeatureFlagsBagProvider` component is now starting after portal is up, which makes the [CompanyModelListener](https://github.com/liferay/liferay-portal/blob/e28af002a4badd99fd6df744074b74845590335f/modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java#L93) activate after the default instance is created. 

# Fix
Make the code that disable the deprecation feature flags re-runnable by having a portal preference to mark a company as processed.

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java [hunk 1]
```diff
@@ -29,29 +36,63 @@ public class CompanyModelListener extends BaseModelListener<Company> {
...
+			_featureFlagsBagProvider.setEnabled(
+				companyId, deprecationFeatureFlag.getKey(), false);
+		}
+
+		portalPreferences.setValue(
+			FeatureFlagConstants.PREFERENCE_NAMESPACE,
+			FeatureFlagConstants.PREFERENCE_KEY, "true");
+
+		_portalPreferencesLocalService.updatePreferences(
+			companyId, PortletKeys.PREFS_OWNER_TYPE_COMPANY, portalPreferences);
```
Comment (ethib137) [inline modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java:86]: Does it make sense to move this code to somewhere shared?

```
		portalPreferences.setValue(
			FeatureFlagConstants.PREFERENCE_NAMESPACE,
			FeatureFlagConstants.PREFERENCE_KEY, "true");

		_portalPreferencesLocalService.updatePreferences(
			companyId, PortletKeys.PREFS_OWNER_TYPE_COMPANY, portalPreferences);
```

**Review by ethib137 — APPROVED** [review]
One suggestion... otherwise it looks ready to send to core infra. Thanks @thiago-buarqque !

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java (hunk 1 above)
Comment (thiago-buarqque) [reply modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java:86]: hmm will do, I'll add it to `FeatureFlagPreferencesManager`, we have it there already but it's private

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:test:upgrade

**Comment by thiago-buarqque** [comment]
ci:test:platform-experience

**Comment by thiago-buarqque** [comment]
ci:stop

File: modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java (hunk 1 above)
Comment (thiago-buarqque) [reply modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java:83]: Let me send a new PR after this one. That way I do it without rushing

**Comment by thiago-buarqque** [comment]
ci:report:59507481

**Comment by thiago-buarqque** [comment]
ci:reevaluate:1307401_10400

**Comment by thiago-buarqque** [comment]
ci:reevaluate:1307401_10400

**Comment by thiago-buarqque** [comment]
There are 3 unique failures. They all failed in a specific environment

`com.liferay.portal.upgrade.test.UpgradeManagerTest` -> `Tomcat 9.0|Chrome 100.0|IBM DB2 11.1|Oracle JDK 8 64-Bit|CentOS 7 64-Bi`

`LocalFile.PortalSmokeUpgrade#ViewPortalSmokeArchiveAndValidateSchema7310` and `LocalFile.PortalSmokeAutoUpgrade#ViewPortalSmokeArchiveAutoUpgrade7210` -> `Tomcat 9.0|Chrome 100.0|PostgreSQL 14.4|Oracle JDK 8 64-Bit|CentOS 7 64-Bit`

There tests are passing in the other seven environments tested. I sent the PR to core-infra https://github.com/liferay-core-infra/liferay-portal/pull/7606

**Pull request closed without merging**